rpcserver.go

```diff
@@ -256,8 +256,14 @@ func (s *rpcServer) serverHandler(blockChan chan int32, blockErrChan chan error)
 					rpcLog.Errorf("Error closing stream: %v",
 						err)
 				}
+
+				// TODO(roasbeef): re-trigger reconnection
```

Comment:
I think what we want to do here instead is create a new `ErrUnexpectedServerShutdown` (the non `io.EOF` case), and then re trigger a reconnection.